packages/create-sitecore-jss/src/templates/nextjs-multisite/scripts/config/plugins/multisite.ts

```diff
@@ -13,19 +14,16 @@ class MultisitePlugin implements ConfigPlugin {
   async exec(config: JssConfig) {
     let sites: SiteInfo[] = [];
 
-    const endpoint = config.graphQLEndpoint;
-    const apiKey = config.sitecoreApiKey;
+    const endpoint = config.sitecoreEdgeContextId ? config.sitecoreEdgeUrl : config.graphQLEndpoint;
 
-    if (!endpoint || !apiKey) {
-      console.warn(
-        chalk.yellow('Skipping site information fetch (missing GraphQL endpoint or API key).')
-      );
+    if (!endpoint) {
+      console.warn(chalk.yellow('Skipping site information fetch (missing GraphQL endpoint).'));
     } else {
       console.log(`Fetching site information from ${endpoint}`);
+
       try {
         const siteInfoService = new GraphQLSiteInfoService({
-          endpoint,
-          apiKey,
+          clientFactory: createGraphQLClientFactory(config),
```

Comment:
FYI: there is no way to import _clientFactory_ since it has access to _temp/config_ that is not generated yet at this stage. Because of that I created a graphql-client-factory/utils to resolve the problem of this plugin

Reply:
Makes sense. Wondering though if we could combine these in a single file (e.g. lib/graphql-client-factory.ts) with both named and default exports?

Reply:
It was my initial solution, but it's not possible, since we have a reference to 'temp/config', so tsc throws a compilation error.
Only one way to workaround that is import 'temp/config' using try/catch :) but it's not the best option and still we need to pass a filled config (createGraphQLClientFactory will throw an error at the first import, since config is empty):
```ts
const getConfig = (): JssConfig => {
  try {
    return require('temp/config');
  } catch (error) {
    return {};
  }
};

// Create a new instance on each import call
export default createGraphQLClientFactory(getConfig());
```

Reply:
Yeah, that's ugly. How about we at least rename "utils" to "create" to make it more fitting?

Reply:
not a problem, sounds better